packages/flutter/lib/src/widgets/scroll_view.dart

```diff
@@ -310,9 +329,24 @@ abstract class ScrollView extends StatelessWidget {
         return buildViewport(context, offset, axisDirection, slivers);
       },
     );
-    return primary && scrollController != null
-      ? PrimaryScrollController.none(child: scrollable)
-      : scrollable;
+    final Widget scrollableResult = primary && scrollController != null
+        ? PrimaryScrollController.none(child: scrollable)
+        : scrollable;
+
+    if (keyboardDismissBehavior == ScrollViewKeyboardDismissBehavior.onDrag) {
+      return NotificationListener<ScrollUpdateNotification>(
+        child: scrollableResult,
+        onNotification: (ScrollUpdateNotification notification) {
+          final FocusScopeNode focusScope = FocusScope.of(context);
```

Comment:
Is there a reason you want to see if the scope is focused, rather than the nearest focus node? It's possible that this `ScrollView` isn't the focused child of the scope: if there's another child of the scope (i.e. a sibling of the `ScrollView`), and it's focused, then `hasFocus` would return true, but children of this `ScrollView` wouldn't have focus.

If you expect the `ScrollView` to be enclosed in a `Focus` widget, you could use `Focus.of(context).hasFocus`. If you don't expect that (which is probably the case), you can introduce a focus node here (probably using a `Focus` widget with `canRequestFocus` set to false) and test to see if that has focus, which would mean that one of the children of the `ScrollView` has focus. You can keep the widget stateless by using a `Builder` to let you look up the node that the `Focus` introduces, as in my previous comment.

Reply:
Here is what I would like to write:
```
if (Keyboard.isVisible) {
  Keyboard.hide();
}
```

If the scrollview that has a sibling that is a TextField that is focused, we want this scrolling action to dismiss his keyboard too.  (I think that is the case you are mentioning).

Reply:
OK, in that case, then looking at the scope is appropriate.